compiler/compiler/src/compiler.rs

```diff
@@ -146,30 +146,50 @@ impl<'a> Compiler<'a> {
     ///
     /// Runs the symbol table pass.
     ///
-    pub fn symbol_table_pass(&self) -> Result<SymbolTable<'_>> {
+    pub fn symbol_table_pass(&'a self) -> Result<SymbolTable> {
         CreateSymbolTable::do_pass((&self.ast, self.handler))
     }
 
     ///
     /// Runs the type checker pass.
     ///
-    pub fn type_checker_pass(&'a self, symbol_table: &mut SymbolTable<'_>) -> Result<()> {
-        TypeChecker::do_pass((&self.ast, &mut symbol_table.clone(), self.handler))
+    pub fn type_checker_pass(&'a self, symbol_table: SymbolTable) -> Result<SymbolTable> {
+        TypeChecker::do_pass((&self.ast, self.handler, symbol_table))
+    }
+
+    ///
+    /// Runs the flattening pass.
+    ///
+    pub fn flattening_pass(&mut self, symbol_table: SymbolTable) -> Result<()> {
+        self.ast = Flattener::do_pass((std::mem::take(&mut self.ast), self.handler, symbol_table))?;
+
+        if self.output_options.ast_initial {
```

Comment:
Should this option be changed to `flattened_ast`?

Reply:
oh! yes thanks for the catch!